### PR TITLE
DoublyLinked_list.cpp

### PR DESCRIPTION
**Pull request by AbhiShingane**
The above program demonstrate the following operations on doubly linkedlist data structure.
1. creation of linked list
2. traversing of linked list(both the direction)
3. Inserting a node at begining of list
4. inserting a node at end of list.
5. Inserting a node after a given node
6. Inserting a node before a given node
7. reversing a doubly linked list.
8. Finding pairs with given sum
9. insert node in sorted way.
10. calculating tiplet with sum equal to value.
11. deleting all occurances of given node.
12. removing all duplicates from sorted linked list.

#### Checklist

- [x] Added description of change
- [ ] Added tests and example, test passes
- [ ] Relevant documentation/comments is changed or added
- [ ] PR title follows semantic [commit guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTION.md#Commit-Guidelines)
- [ ] Search previous suggestions before making a new one, as yours may be a duplicate.
- [ ] Sort by alphabetical order
- [x] I acknowledge that all my contributions will be made under the project's license.

Notes:

### CODE REVIEW

**Comment by bhaumikmistry** [comment]
- [ ] Please check in one data structure at a time
- [ ] Convert tree to list should not be placed in the  structure directory
- [ ] Document you code for easy to use, add function definitions, calling parameter node and hnode will not help any new learner to understand. 
- [ ] Make a class and then a different class or main() for a user input and api, for example your double liked list can not be reused by anyone, since it prints and asks for user input within link list class

Best.

File: Data Structure/CircularSinglyLinkedList.cpp [hunk 1]
```diff
@@ -0,0 +1,296 @@
+#include <stdio.h>
```
Comment (kvedala) [inline Data Structure/CircularSinglyLinkedList.cpp:1]: This is a pure C program, not a C++

File: Data Structure/ConvertTreeToList.cpp [hunk 2]
```diff
@@ -0,0 +1,87 @@
+#include <iostream>
+using namespace std;
```
Comment (kvedala) [inline Data Structure/ConvertTreeToList.cpp:2]: 1. cannot use `using namespace` 
2. missing documentation in the code

File: Data Structure/DoublyLinked_list.cpp [hunk 3]
```diff
@@ -0,0 +1,461 @@
+// program/operations  on doubly linked list
+
```
Comment (kvedala) [inline Data Structure/DoublyLinked_list.cpp:2]: Also a pure C file, not a C++

**Pull request closed without merging**